src/app/applications/applications.directive.upload.js

```diff
@@ -0,0 +1,91 @@
+(function () {
+	'use strict';
+
+	angular
+		.module( 'roboconf.applications' )
+		.directive( 'rbcfZipUpload', rbcfZipUpload );
+
+	/* @ngInject */
+	function rbcfZipUpload() {
+		return {
+			restrict: 'E',
+			templateUrl: 'templates/applications/_upload-directive.html',
+			controller: rbcfZipUploadController
+		};
+	}
+
+	rbcfZipUploadController.$inject = [ 'rPrefs', '$scope', 'rShare' ];
+	function rbcfZipUploadController( rPrefs, $scope, rShare ) {
+		
+		// Fields and functions
+		$scope.restUrl = rPrefs.getUrl() + '/applications/templates';
+		$scope.uploadZip = uploadZip;
+		
+		// Functions
+		function uploadZip() {
+			
+			// Prepare the data to submit
+			var formObj = $( '#new-app-form' )[ 0 ];
+			var formData = new FormData( formObj );
+				
+			// Create an ajax request
+			$.ajax({
+				// This is to follow upload progression
+				xhr: function() {
+			        var xhr = new window.XMLHttpRequest();
+			        xhr.upload.addEventListener('progress', function(evt) {
+			            if( evt.lengthComputable ) {
+			                var percentComplete = Math.round((evt.loaded * 100) / evt.total);
+			                updateProgressBar( percentComplete );
+			            }
+			       }, false);
+
+			       return xhr;
+			    },
+			    
+			    // Other properties
+				url: $scope.restUrl,
+				type: 'POST',
+				data: formData,
+				mimeType: 'multipart/form-data',
+				contentType: false,
+				cache: false,
+				processData: false,
+				
+				// Callbacks on success or failure
+				success: onSuccessfulUpload,
+				error : onFailingUpload
+			});
+		}
+		
+		function updateProgressBar( percentage ) {
```

Comment:
That's exactly what I did.
For the record, I took this from a code snippet on the internet. But since I already reworked it a little bit, I could push it farther and deal with the progress bar differently.